Sources/ShapedArray/Initializers.swift

```diff
@@ -0,0 +1,71 @@
+//
+//  Initializers.swift
+//  
+//
+//  Created by Jaap Wijnen on 19/08/2022.
+//
```

Comment:
Similar to my comment about having operations in separate files, I see even less a reason to have initializers in files separate from the definition of the type and its required initializer. It adds more friction to reading the code and I can’t think of a positive especially since we’ve already split some types off into their own files to lessen loc.

Reply:
I copied this from s4tf's approach since (gauging from their codebase) there will be an enormous amount of initialisers once the library gets rounded out more.

Reply:
I think that approach makes sense. Keep the declaration and protocol conformances in one file. Complex initializers and methods go elsewhere. However, we could wait to create a separate file until we have multiple initializers.

Reply:
Ok, if there are a lot of initializers I can get behind this approach :+1: